repo_full_name,issue_id,comment_id,user_login,user_id,created_at,updated_at,body,reactions_total,reactions_plus1,reactions_minus1,reactions_laugh,reactions_hooray,reactions_confused,reactions_heart
ipython/ipython,3049206975,2863316258,krassowski,5832902,2025-05-08T14:43:38+00:00,2025-05-08T14:54:22+00:00,"The reasoning here is:
- if we treat non-system exceptions in `%%time` as something that user may have done intentionally (i.e. `%%time` is a sandbox for measuring execution time of code, whether it raises or not), then we should show the execution time after catching the exception (which is not the case)
- if we treat non-system exceptions in `%%time` as something which user did not intend to occur, then we should stop execution (which is not the case either).

Neither solution will make all users happy, but we already have a pattern which reconciliates these two stances: the `%%script` magic (and its derivatives) has a `--no-raise-error` argument which allows user to communicate the intended behaviour:

https://github.com/ipython/ipython/blob/637e037d668db7aebe61ae04261ebfc5cb7f32c5/IPython/core/magics/script.py#L58-L62

Whether user expects the error or not, we should still re-raise `KeyboardInterrupt` as that is used by the stop button.",0,0,0,0,0,0,0
ipython/ipython,3045233000,2868773651,Carreau,335567,2025-05-10T11:42:14+00:00,2025-05-10T11:42:14+00:00,ctrl-o should force a new line. ,0,0,0,0,0,0,0
